response_operations_ui/views/update_event_date.py

```diff
@@ -66,17 +66,33 @@ def update_event_date_submit(short_name, period, tag):
         abort(404)
 
     if not form.validate() or not valid_date_for_event(tag, form):
-        return update_event_date(short_name, period, tag, errors=form.errors)
-
-    submitted_dt = datetime(year=int(form.year.data),
-                            month=int(form.month.data),
-                            day=int(form.day.data),
-                            hour=int(form.hour.data),
-                            minute=int(form.minute.data),
-                            tzinfo=tz.gettz('Europe/London'))
-    if not collection_exercise_controllers.update_event(exercise['id'], tag, submitted_dt):
+        for error in form.errors.values():
+            flash(error, 'error')
         return redirect(url_for('collection_exercise_bp.update_event_date',
-                                short_name=short_name, period=period, tag=tag, errors=True))
+                                short_name=short_name, period=period, tag=tag))
 
+    try:
+        submitted_dt = datetime(year=int(form.year.data),
+                                month=int(form.month.data),
+                                day=int(form.day.data),
+                                hour=int(form.hour.data),
+                                minute=int(form.minute.data),
+                                tzinfo=tz.gettz('Europe/London'))
+
+        error_message = collection_exercise_controllers.update_event(
+            collection_exercise_id=exercise['id'], tag=tag, timestamp=submitted_dt)
+        if error_message is not None:
+            flash(error_message, 'error')
+
+    except ValueError as exc:
+        error_message = str(exc)
+
+    if error_message:
+        for error in form.errors.values():
+            if error is not None:
+                flash(error, 'error')
+        return redirect(url_for('collection_exercise_bp.update_event_date',
+                                short_name=short_name, period=period, tag=tag))
+    success_panel = "Event date updated."
```

Comment:
Don't really need to make a variable of this, should be able to just put it into the url_for as it is.